odoo/orm/fields_textual.py

```diff
@@ -86,6 +86,8 @@ def convert_to_column(self, value, record, values=None, validate=True):
 
     def convert_to_column_insert(self, value, record, values=None, validate=True):
         if self.translate:
+            if isinstance(value, dict):
```

Comment:
The orm won't accept this at least for 19.0.
If we allow the write value of translated field to be a dict. Any one who override `create` or `write` methods and want to add some extra logics have to check `if isinstance(value, dict):` everytime.

If you really want similar feature do something like `_pre/post_create` for the model you want to hack